files/default/ext_auth_files/pcluster_dcv_ext_auth.py

```diff
@@ -0,0 +1,362 @@
+        return json.dumps({"sessionToken": session_token})
+
+    @staticmethod
+    def _sanitize_request(string_to_test, regex, message):
+        if not re.match(regex, string_to_test):
+            raise ExtAuthHandler.IncorrectRequestException(message)
+
+    @staticmethod
+    def _is_session_valid(user, session_id):
+        processes = subprocess.check_output(["ps", "aux"]).decode("utf-8").split("\n")
+        if not filter(lambda x: ExtAuthHandler.is_valid_process(x, user, session_id), processes):
+            raise ExtAuthHandler.IncorrectRequestException("The given session for the user does not exists")
+
+    @staticmethod
+    def _verify_session_existence(user, session_id):
+        retry_call(ExtAuthHandler._is_session_valid, fargs=[user, session_id], tries=5, delay=1)
+
+    @staticmethod
+    def is_valid_process(row, user, session_id):
+        fields = row.split()
+        command_index = 10
+        session_name_index = 12
+        user_index = 0
+        dcv_agent_path = "/usr/libexec/dcv/dcvagent"
+        return (
+                fields[command_index] == dcv_agent_path
+                and fields[user_index] == user
+                and fields[session_name_index] == session_id
+        )
+
+
+class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
+    """Handle requests in a separate thread."""
+
+
+def set_server_logging(server):
+    log_file = open(LOG_FILE_PATH, "w")
+    os.chmod(LOG_FILE_PATH, 0o644)
+    server.log_file = log_file
+
+
+def run_server(port, certificate=None, key=None):
+    """
+    This class run the external authenticator server on localhost.
+
+    The external authenticator *must* be run as a separate user, and the script *must* be executed in the home folder of
+    the user. The user has to have in his home an empty folder, /.accepted_sessions and an executable file with the
+    setuid bit set, write_token.exe.
+
+    :param port: the port in which you want to start the server
+    :param certificate: the certificate to use if https
+    :param key: the key to use if https and it's not in certificate
+    """
+    server_class = ThreadedHTTPServer
+    handler_class = ExtAuthHandler
+    server_address = ("localhost", port)
+    httpd = server_class(server_address, handler_class)
+    # the following error is not real.
+    set_server_logging(httpd)
+    if certificate:
+        if key:
+            httpd.socket = ssl.wrap_socket(httpd.socket, certfile=certificate, keyfile=key, server_side=True)
+        else:
+            httpd.socket = ssl.wrap_socket(httpd.socket, certfile=certificate, server_side=True)
+    print(
+        "Starting DCV external authenticator {PROTOCOL} server on port {PORT}, use <Ctrl-C> to stop".format(
+            PROTOCOL="HTTPS" if certificate else "HTTP", PORT=port
+        )
+    )
+    httpd.serve_forever()
+
+
+def add_argument(parser, name, description, expected_type=str):
+    parser.add_argument(name, help=description, type=expected_type)
```

Comment:
this function is not really helpful, for some reason:
* it is a single line function so we are not saving code lines
* you have to pass to the add_argument more parameters than the parser.add_argument one
* the function is doing the same thing of the parser.add_argument without any advantage

In cases like the one described before we can avoid to define a new function.